src/internal/pfsdb/commits.go

```diff
@@ -0,0 +1,953 @@
+package pfsdb
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"github.com/pachyderm/pachyderm/v2/src/internal/coredb"
+	"github.com/pachyderm/pachyderm/v2/src/internal/dbutil"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"strings"
+	"time"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/internal/stream"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	// CommitsChannelName is used to watch events for the commits table.
+	CommitsChannelName = "pfs_commits"
+	createCommit       = `
+		WITH repo_row_id AS (SELECT id from pfs.repos WHERE name=:repo.name AND type=:repo.type AND project_id=(SELECT id from core.projects WHERE name= :repo.project.name))
+		INSERT INTO pfs.commits 
+    	(commit_id, 
+    	 commit_set_id, 
+    	 repo_id, 
+    	 branch_id, 
+    	 description, 
+    	 origin, 
+    	 start_time, 
+    	 finishing_time, 
+    	 finished_time, 
+    	 compacting_time_s, 
+    	 validating_time_s, 
+    	 size, 
+    	 error) 
+		VALUES 
+		(:commit_id, 
+		 :commit_set_id, 
+		 (SELECT id from repo_row_id), 
+		(SELECT id from pfs.branches WHERE name=:branch_name AND repo_id=(SELECT id from repo_row_id)), 
+		:description, 
+		:origin, 
+		:start_time,
+		:finishing_time,
+		:finished_time,
+		:compacting_time_s,
+		:validating_time_s,
+		:size,
+		:error)
+		RETURNING int_id;`
+	updateCommit = `
+		WITH repo_row_id AS (SELECT id from pfs.repos WHERE name=:repo.name AND type=:repo.type AND project_id=(SELECT id from core.projects WHERE name= :repo.project.name))
+		UPDATE pfs.commits SET 
+			commit_id=:commit_id, 
+			commit_set_id=:commit_set_id,
+		    repo_id=(SELECT id from repo_row_id), 
+		    branch_id=(SELECT id from pfs.branches WHERE name=:branch_name AND repo_id=(SELECT id from repo_row_id)), 
+			description=:description, 
+			origin=:origin, 
+			start_time=:start_time, 
+			finishing_time=:finishing_time, 
+			finished_time=:finished_time, 
+		    compacting_time_s=:compacting_time_s, 
+			validating_time_s=:validating_time_s, 
+			size=:size, 
+			error=:error 
+		WHERE int_id=:int_id;`
+	getCommit = `
+		SELECT 
+    		commit.int_id, 
+    		commit.commit_id, 
+    		commit.repo_id, 
```

Comment:
This way you can deserialize into the embedded `Commit.Repo`

```suggestion
    		commit.repo_id as "repo.id",
```